crates/aptos-metrics-core/src/const_metric.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use prometheus::{
+    core::{Collector, Desc},
+    proto::{Counter, Gauge, LabelPair, Metric, MetricFamily, MetricType},
+};
+
+/// Provides a metric with one fixed value that cannot be changed. Users of this
+/// package will not have much use for it in regula operations. However, when
+/// implementing custom Collectors, it is useful as a throw-away metric that is
+/// generated on the fly to send it to Prometheus in the Collect method.
+/// Reference: https://github.com/prometheus/client_golang/blob/main/prometheus/value.go#L106
+pub struct ConstMetric {
+    desc: Desc,
+    metric: Metric,
+    metric_type: MetricType,
+}
+
+impl ConstMetric {
+    pub fn new_counter(
+        desc: Desc,
+        value: f64,
+        label_values: Option<&[String]>,
+    ) -> Result<ConstMetric> {
+        let label_values = label_values.unwrap_or_default();
+        if desc.variable_labels.len() != label_values.len() {
```

Comment:
Should you be checking for the actual labels as well?

Reply:
Not unless label_values also includes the label along with the value. The assumption here is that the label_values are in the same order as the labels, which is the standard assumption in the prometheus crate.